rskj-core/src/main/java/co/rsk/peg/BridgeSerializationUtils.java

```diff
@@ -61,19 +61,51 @@ private BridgeSerializationUtils() {
         throw new IllegalAccessError("Utility class, do not instantiate it");
     }
 
-    public static byte[] serializeRskTxWaitingForSignatures(
-          Map.Entry<Keccak256, BtcTransaction> rskTxWaitingForSignaturesEntry) {
-        byte[][] serializedRskTxWaitingForSignaturesEntry =
-            serializeRskTxWaitingForSignaturesEntry(rskTxWaitingForSignaturesEntry);
+    public static byte[] serializeRskTxHash(Keccak256 rskTxHash) {
+        return RLP.encodeElement(rskTxHash.getBytes());
+    }
+
+    private static Keccak256 deserializeRskTxHashFromRLPData(byte[] rlpData) {
+        return new Keccak256(rlpData);
+    }
+
+    public static byte[] serializeBtcTransaction(BtcTransaction btcTransaction) {
+        return RLP.encodeElement(btcTransaction.bitcoinSerialize());
+    }
+
+    public static BtcTransaction deserializeBtcTransaction(byte[] data, NetworkParameters networkParameters, boolean txHasInputs) {
+        if (data == null || data.length == 0) {
+            return null;
+        }
+
+        byte[] rlpData = RLP.decode2(data).get(0).getRLPData();
+        return deserializeBtcTransactionFromRLPData(rlpData, networkParameters, txHasInputs);
+    }
+
+    private static BtcTransaction deserializeBtcTransactionFromRLPData(byte[] rlpData, NetworkParameters networkParameters, boolean txHasInputs) {
+        if (!txHasInputs) {
+            BtcTransaction tx = new BtcTransaction(networkParameters);
+            tx.parseNoInputs(rlpData);
+            return tx;
+        }
+
+        return new BtcTransaction(networkParameters, rlpData);
+    }
```

Comment:
```suggestion
  private static BtcTransaction deserializeBtcTransactionFromRLPData(
        byte[] rlpData, NetworkParameters networkParameters, boolean hasInputs) {
    
    if (hasInputs) {
        return new BtcTransaction(networkParameters, rlpData);
    }
    
    BtcTransaction tx = new BtcTransaction(networkParameters);
    tx.parseNoInputs(rlpData);
    return tx;
}
```

Reply:
i prefer to have the less used case inside the if, to improve readability.
(Currently, we only use this deserialization having inputs)